docs/architecture/ADR-004-IBC-whitelist.md

```diff
@@ -0,0 +1,27 @@
+# ADR 004: IBC Whitelist/Allowlist
```

Comment:
[optional] one term is sufficient to describe this. Slight preference for `Allowlist` because https://github.com/rails/rails/issues/33677

```suggestion
# ADR 004: IBC Allowlist
```

Note: accepting this suggestion implies updating the filename too. Preferably: `adr-004-ibc-allowlist.md` to conform to https://github.com/celestiaorg/celestia-app/pull/772